python/lsst/sphgeom/_yaml.py

```diff
@@ -56,3 +61,53 @@ def region_constructor(loader, node):
 
 for loader in YamlLoaders:
     yaml.add_constructor("lsst.sphgeom.Region", region_constructor, Loader=loader)
+
+
+# Pixelization schemes
+
+def pixel_representer(dumper, data, pixelScheme):
+    """Represent a pixelization in YAML
+
+    Stored as the pixelization level. Use a mapping for more clarity.
+    """
+    return dumper.represent_mapping(f"lsst.sphgeom.{pixelScheme}",
+                                    {"level": data.getLevel()})
+
+
+def pixel_constructor(loader, node, cls):
+    mapping = loader.construct_mapping(node)
+    return cls(mapping["level"])
+
+
+def makeYamlPixels(pixelSchemeCls):
+    """Given a pixelization class, return YAML representer and
+    constructor.
+
+    Parameters
+    ----------
+    pixelSchemeCls : `type`
+        The pixelization class.
+
+    Returns
+    -------
+    representer : function
+        Function to be used as a yaml representer.
+    constructor : function
+        Function to be used as a yaml constructor.
+    """
+    def representer(dumper, data):
+        return pixel_representer(dumper, data, pixelSchemeCls.__name__)
+
+    def constructor(loader, node):
+        return pixel_constructor(loader, node, pixelSchemeCls)
+
+    return representer, constructor
+
+
+# All the pixelization schemes use the same approach with getLevel
+for pixelSchemeCls in (HtmPixelization, Q3cPixelization, Mq3cPixelization):
+    representer, constructor = makeYamlPixels(pixelSchemeCls)
+
+    yaml.add_representer(pixelSchemeCls, representer)
+    for loader in YamlLoaders:
+        yaml.add_constructor(f"lsst.sphgeom.{pixelSchemeCls.__name__}", constructor, Loader=loader)
```

Comment:
Will this be a problem if there's a pixelization class that's not at the `sphgeom` top level? It might be better to get the fully qualified name, then use that here and for `pixel_representer`.

Reply:
It works for what we have now and it doesn't look like healpix should be implemented differently. If it does we can change it later.